crates/oxc_mangler/src/lib.rs

```diff
@@ -387,12 +471,12 @@ impl Mangler {
         keep_name_symbols: &FxHashSet<SymbolId>,
         total_number_of_slots: usize,
         slots: &[Slot],
-        allocator: &'a Allocator,
     ) -> Vec<'a, SlotFrequency<'a>> {
         let root_scope_id = scoping.root_scope_id();
         let mut frequencies = Vec::from_iter_in(
-            repeat_with(|| SlotFrequency::new(allocator)).take(total_number_of_slots),
-            allocator,
+            repeat_with(|| SlotFrequency::new(self.temp_allocator.as_ref()))
+                .take(total_number_of_slots),
+            self.temp_allocator.as_ref(),
```

Comment:
Probably preferable to hoist `let temp_allocator = self.temp_allocator.as_ref();` to top of the function. Not sure if compiler is smart enough to do that for us - it may end up executing `self.temp_allocator.as_ref()` repeatedly for every call to the `repeat_with` closure.